frontend/packages/dev-console/src/utils/pipeline-filter-reducer.ts

```diff
@@ -58,3 +58,8 @@ export const pipelineResourceTypeFilter = (filters, pipelineResource): boolean =
   const type = pipelineResourceFilterReducer(pipelineResource);
   return filters.selected.has(type) || !_.includes(filters.all, type);
 };
+
+export const taskRunFilterReducer = (taskRun): string => {
+  const status = pipelineRunStatus(taskRun);
```

Comment:
There's one issue with that. It's a cyclic dependence and pipeline-filter-reducer doesn't have types. It should be a separate issue to refactor `Types` out of pipeline-augment. cc: @andrewballantyne

Reply:
I would prefer here to not reuse `pipelineRunStatus` at all and to not mix PipelineRun and TaskRun objects. I personally would prefer in this case "duplicated" code for both types instead of methods which supports both.

Reply:
Yeah, types are annoying in this file. So it should be handled by the work to take Pipelines out of DevConsole package